### ISSUE TITLE
Deprecate UseHttp2 in javadsl

### ISSUE THREAD

**Issue by patriknw** [akka/akka-http#2705]
In scaladsl it's deprecated and it works fine without it. In Java it's not deprecated and it doesn't work without it. Have to use 
```
ConnectHttp.toHost(settings.hostPort.hostname, settings.hostPort.port, UseHttp2.always())
```
for Java (which isn't indicated in docs https://doc.akka.io/docs/akka-grpc/current/server/walkthrough.html#serving-the-service-with-akka-http)

**Comment by jrudolph** [comment]
Could you look at that @raboof?

**Comment by raboof** [comment]
@patriknw https://github.com/akka/akka-samples/pull/125/files is using akka-grpc 0.6.2 and akka-http 10.1.8 - I think this was already fixed in akka-grpc 0.7.0 and akka-http 10.1.9.

I'd be happy to update that PR and test if that indeed fixes it there as well, how did you test it?

**Comment by patriknw** [comment]
Sorry, I had an old Akka gRPC 0.6.2. Updated and it works as expected.
